app/simulation_test.go

```diff
@@ -86,28 +98,107 @@ func BenchmarkSimulation(b *testing.B) {
 		simapp.EmptyAppOptions{},
 	)
 
-	simApp, ok := app.(SimApp)
+	app, ok := cmdApp.(SimApp)
 	require.True(b, ok, "can't use simapp")
 
 	// Run randomized simulations
 	_, simParams, simErr := simulation.SimulateFromSeed(
 		b,
 		os.Stdout,
-		simApp.GetBaseApp(),
-		simapp.AppStateFn(simApp.AppCodec(), simApp.SimulationManager()),
+		app.GetBaseApp(),
+		simapp.AppStateFn(app.AppCodec(), app.SimulationManager()),
 		simulationtypes.RandomAccounts,
-		simapp.SimulationOperations(simApp, simApp.AppCodec(), config),
-		simApp.ModuleAccountAddrs(),
+		simapp.SimulationOperations(app, app.AppCodec(), config),
+		app.ModuleAccountAddrs(),
 		config,
-		simApp.AppCodec(),
+		app.AppCodec(),
 	)
 
 	// export state and simParams before the simulation error is checked
-	err = simapp.CheckExportSimulation(simApp, config, simParams)
+	err = simapp.CheckExportSimulation(app, config, simParams)
 	require.NoError(b, err)
 	require.NoError(b, simErr)
 
 	if config.Commit {
 		simapp.PrintStats(db)
 	}
 }
+
+func TestAppStateDeterminism(t *testing.T) {
```

Comment:
This is very interesting, I thought incorrect `apphash` was raised in the classic simulation testing we have.

If this is relevant, we can maybe raise an issue on `starport` to include this into the app scaffolding since we already scaffold simulation testing

Reply:
You need to run the same set of transactions (basically just using the same seed) multiple times and save the app hash of the previous run to check it against the current state.

There are a lot more simulation options used in the [Cosmos SDK repo](https://github.com/cosmos/cosmos-sdk/blob/master/simapp/sim_test.go) that gave me inspiration.  

I think the non-determinism testing is one of the most important parts of having simulation testing, so it's probably relevant to raise an issue on `starport`, yes.